compatibility.bs

```diff
@@ -832,6 +835,136 @@ supported on all {{Window}} objects and <code><{body}></code> elements as attrib
+
+The <dfn>redirect heuristic grant duration</dfn> is an [=implementation-defined=]
+[=duration=] that represents the length of time after the redirect heuristic is
+detected that the bounced [=host=] will have access to the final [=host=].
+
+Note: 15 minutes is a reasonable [=redirect heuristic grant duration=] value.
+
+<h3 id="sa-heuristics-grant">Storage Access Grant</h3>
+
+Define a [=powerful feature=] identified by the [=powerful feature/name=]
+"storage-access-heuristics", with the following <dfn>permission key type</dfn>:
+
+A [=permission key=] of the "storage-access-heuristics" feature is a [=tuple=] consisting
+of a [=host=] <dfn property>top-level</dfn> and a [=host=] <dfn property>requester</dfn>.
+
+<div algorithm>
+
+To <dfn>grant access for heuristics</dfn> given a [=host=] |host|, [=host=]
+|firstPartyHost|, and [=duration=] |duration|, perform the following steps:
+
+1. Let |key| be a [=permission key=] with <var ignore>top-level</var> set to |firstPartyHost|
+    and <var ignore>requester</var> set to |host|.
+1. Queue a task on the [=current settings object=]'s [=responsible event loop=] to:
+    1. [=Set a permission store entry=] with:
+      <dl>
+          <dt><var ignore>descriptor</var></dt>
+          <dd>"storage-access-heuristics"</dd>
+          <dt><var ignore>key</var></dt>
+          <dd>|key|</dd>
+          <dt><var ignore>current state</var></dt>
+          <dd>"granted"</dd>
+      </dl>
+    2. Set the new permission's [=permission/lifetime=] to |duration|.
+
+<div class=note>
+This algorithm is based on [=request permission to use=], except for the following key differences:
+- It sets a dynamic permission [=permission/lifetime=].
+- It generates a [=permission key=] independently of the [=current settings object=].</div>
+
+</div>
+
+<h3 id="sa-heuristics-popup">Popup Heuristic</h3>
+
+Append the following steps to the <a spec="html">activation notification</a>
+steps in the [[HTML#user-activation-processing-model|user activation processing
+model]]:
+
+1. Run [=detect a popup heuristic=] given <var ignore>document</var>.
+
+<div algorithm>
+
+To <dfn>detect a popup heuristic</dfn> given a [=Document=] |document|,
+perform the following steps:
+
+1. Let |browsingContext| be |document|'s [=Document/browsing context=].
+1. If |browsingContext|'s <var ignore>is popup</var> is false, then abort these steps.
+1. If |browsingContext|'s <var ignore>opener origin at creation</var> is null, then abort these steps.
+1. Let |navigable| be |document|'s [=node navigable=].
+1. If |navigable| is null, then abort these steps.
+1. Let |topDocument| be |navigable|'s [=top-level traversable=]'s
+    [=navigable/active document=].
+1. Let |origin| be |topDocument|'s [=Document/origin=].
+1. If |origin| is an [=opaque origin=] then abort these steps.
+1. Let |site| be the result of running [=obtain a site=] given |origin|.
+1. Let |host| be |site|'s [=host=].
+1. [=Grant access for heuristics=] given:
```

Comment:
Filed an issue in my personal repo, since it has the explainer: https://github.com/amaliev/3pcd-exemption-heuristics/issues/3. Let me know if I should move it to this one.

Reply:
If this is to be merged in the actual Web Compat standard, then yeah this issue should probably move over to that repo. But maybe Mike can make the final call, since he's the editor here.

Reply:
I'm still not sure this should be in the compat spec.